content/en/docs/languages/dotnet/metrics/best-practices.md

```diff
@@ -0,0 +1,451 @@
+semantics and performance. Check the
+[Instrument Selection](/docs/specs/otel/metrics/supplementary-guidelines#instrument-selection)
+section from the supplementary guidelines for more information.
+
+> {{% /alert %}}
+
+| OpenTelemetry Specification                                                            | .NET Instrument Type                                                                                                        |
+| -------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------- |
+| [Asynchronous Counter](/docs/specs/otel/metrics/api/#asynchronous-counter)             | [`ObservableCounter<T>`](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.observablecounter-1)             |
+| [Asynchronous Gauge](/docs/specs/otel/metrics/api/#asynchronous-gauge)                 | [`ObservableGauge<T>`](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.observablegauge-1)                 |
+| [Asynchronous UpDownCounter](/docs/specs/otel/metrics/api/#asynchronous-updowncounter) | [`ObservableUpDownCounter<T>`](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.observableupdowncounter-1) |
+| [Counter](/docs/specs/otel/metrics/api/#counter)                                       | [`Counter<T>`](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.counter-1)                                 |
+| [Gauge](/docs/specs/otel/metrics/api/#gauge)                                           | [`Gauge<T>`](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.gauge-1)                                     |
+| [Histogram](/docs/specs/otel/metrics/api/#histogram)                                   | [`Histogram<T>`](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.histogram-1)                             |
+| [UpDownCounter](/docs/specs/otel/metrics/api/#updowncounter)                           | [`UpDownCounter<T>`](https://learn.microsoft.com/dotnet/api/system.diagnostics.metrics.updowncounter-1)                     |
+
+Avoid creating instruments (for example, `Counter<T>`) too frequently.
+Instruments are fairly expensive and meant to be reused throughout the
+application. For most applications, instruments can be modeled as static
+readonly fields or singleton through dependency injection.
+
+Avoid invalid instrument names.
+
+> {{% alert title="Note" %}} OpenTelemetry will not collect metrics from
+> instruments that are using invalid names. Refer to the
+> [OpenTelemetry Specification](/docs/specs/otel/metrics/api/#instrument-name-syntax)
+> for the valid syntax. {{% /alert %}}
+
+Avoid changing the order of tags while reporting measurements. For example:
+
+> {{% alert title="Warning" color="warning" %}} The last line of code has bad
+> performance since the tags are not following the same order: {{% /alert %}}
```

Comment:
```suggestion
{{% alert title="Warning" color="warning" %}} The last line of code has bad
performance since the tags are not following the same order: {{% /alert %}}
```